domain/user/state/state.go

```diff
@@ -291,6 +291,95 @@ func (st *State) SetPasswordHash(ctx context.Context, uuid user.UUID, passwordHa
 	})
 }
 
+// EnableUserAuthentication will enable the user with the supplied UUID. If the user does not
+// exist an error that satisfies usererrors.NotFound will be returned.
+func (st *State) EnableUserAuthentication(ctx context.Context, uuid user.UUID) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Annotate(err, "getting DB access")
+	}
+
+	enableUserQuery := `
+UPDATE user_authentication
+SET disabled = false
+WHERE user_uuid = $M.uuid
+`
+
+	updateEnableUserStmt, err := sqlair.Prepare(enableUserQuery, sqlair.M{})
+	if err != nil {
+		return errors.Annotate(err, "preparing update enableUserAuthentication query")
+	}
+
+	return db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		removed, err := st.isRemoved(ctx, tx, uuid)
+		if err != nil {
+			return errors.Annotatef(err, "getting user with uuid %q", uuid)
+		}
+		if removed {
+			return errors.Annotatef(usererrors.NotFound, "%q", uuid)
+		}
+
+		err = ensureUserAuthentication(ctx, tx, uuid)
+		// If the user does not exist then we return the error as is.
+		// If the user does exist but their authentication is disabled then we
+		// continue on to enable the user.
+		if err != nil && !errors.Is(err, usererrors.UserAuthenticationDisabled) {
+			return errors.Annotatef(err, "authenticating user with uuid %q", uuid)
+		}
+
+		query := tx.Query(ctx, updateEnableUserStmt, sqlair.M{"uuid": uuid.String()})
+		err = query.Run()
+		if err != nil {
+			return errors.Annotatef(err, "enabling user authentication with uuid %q", uuid)
+		}
+
+		return nil
+	})
+}
+
+// DisableUserAuthentication will disable the user with the supplied UUID. If the user does
+// not exist an error that satisfies usererrors.NotFound will be returned.
+func (st *State) DisableUserAuthentication(ctx context.Context, uuid user.UUID) error {
+	db, err := st.DB()
+	if err != nil {
+		return errors.Annotate(err, "getting DB access")
+	}
+
+	disableUserQuery := `
+UPDATE user_authentication
```

Comment:
Same comment above with the upsert. Can avoid a double trip and the call to ensureUserAuthentication if you want.